lampod-cli/src/main.rs

```diff
@@ -112,15 +158,19 @@ fn run(args: LampoCliArgs) -> error::Result<()> {
         radicle_term::success!("Wallet Generated, please store these words in a safe way");
         radicle_term::println(
             radicle_term::format::badge_primary("wallet-keys"),
-            format!("{}", radicle_term::format::highlight(mnemonic)),
+            format!("{}", radicle_term::format::highlight(mnemonic.clone())),
         );
+        write_words_to_file(format!("{}/wallet.dat", words_path), &Some(mnemonic))?;
         wallet
     } else {
         match client.kind() {
             lampo_common::backend::BackendKind::Core => {
                 // SAFETY: It is safe to unwrap the mnemonic because we check it
                 // before.
-                CoreWalletManager::restore(Arc::new(lampo_conf.clone()), &mnemonic.unwrap())?
+                CoreWalletManager::restore(
+                    Arc::new(lampo_conf.clone()),
+                    &mnemonic.clone().unwrap(),
```

Comment:
```suggestion
                    &mnemonic.clone().unwrap(),
```

Why we are calling `unwrap()`?

Reply:
because mnemonic is of type `Option<string>` here

Reply:
Mh, but the mnemonic should be always specified at this point right? otherwise lampo crash.

So there is a way to set the mnemonic to `String` and not `Option`?

Reply:
This can be implement. But we do have a comment there already

```
lampo_common::backend::BackendKind::Core => {
                // SAFETY: It is safe to unwrap the mnemonic because we check it
                // before.
                CoreWalletManager::restore(
                    Arc::new(lampo_conf.clone()),
                    &mnemonic.clone().unwrap(),
                )?
            }
```

Reply:
Yeah this could works but the code will be not sound, there is no reason to arrived here with the mnemonic still of type `Option`. See https://github.com/vincenzopalazzo/lampo.rs/pull/295#discussion_r1823144469